src/gt4py/analysis/passes.py

```diff
@@ -456,57 +472,194 @@ def __init__(self):
     def defaults(self):
         return self._DEFAULT_OPTIONS
 
-    def apply(self, transform_data: TransformData):
-        # Greedy strategy to merge multi-stages
-        merged_blocks = [transform_data.blocks[0]]
-        for candidate in transform_data.blocks[1:]:
-            merged = merged_blocks[-1]
-            if self._are_compatible_multi_stages(
-                merged, candidate, transform_data.has_sequential_axis
-            ):
-                merged.id = transform_data.id_generator.new
-                self._merge_domain_blocks(merged, candidate)
+    def _greedy_merging(
+        self,
+        items: Sequence[T],
```

Comment:
Nice to see this being completely generic! Optional, but how about a small docstring explaining that item is _usually_ a block or an ij block?